Demo/Tests/MoyaProviderSpec.swift

```diff
@@ -7,7 +7,7 @@ class MoyaProviderSpec: QuickSpec {
     override func spec() {
         var provider: MoyaProvider<GitHub>!
         beforeEach {
-            provider = MoyaProvider<GitHub>(stubClosure: MoyaProvider.ImmediatelyStub)
+            provider = MoyaProvider<GitHub>(stubClosure: MoyaDefaults.ImmediatelyStub)
```

Comment:
I'm really liking this PR!  However, I'm gonna ask for a few changes to go into a separate PR, like the change from `MoyaProvider` to `MoyaDefaults`.  This is a breaking change, and I'm hoping we can support structs without any breaking changes.